### ISSUE TITLE
`allow_major_version_upgrade` is missing from aws_dms_replication_instance

### ISSUE THREAD

**Issue by eoliphan** [terraform-providers/terraform-provider-aws#14438]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The AllowMajorVersionUpgrade flag is necessary to perform major updates to the aws_dms_replication_instance type

### New or Affected Resource(s)

aws_dms_replication_instance

### Potential Terraform Configuration

```hcl
resource "aws_dms_replication_instance" "default" {
  allocated_storage            = var.allocated_storage
  apply_immediately            = true
  auto_minor_version_upgrade   = true
  engine_version               = var.engine_version
  allow_major_version_upgrade = true
```

### References

* [ModifyReplicatonInstance docs](https://docs.aws.amazon.com/dms/latest/APIReference/API_ModifyReplicationInstance.html#API_ModifyReplicationInstance_RequestSyntax)

**Comment by bflad** [comment]
Adding this functionality would also allow fixing the currently failing test:

```
=== CONT  TestAccAWSDmsReplicationInstance_EngineVersion
TestAccAWSDmsReplicationInstance_EngineVersion: testing.go:684: Step 1 error: errors during apply:
Error: error modifying DMS Replication Instance (tf-acc-test-1816079488213827146): InvalidParameterCombinationException: The AllowMajorVersionUpgrade flag must be present when upgrading to a new major version.
```

**Comment by bflad** [comment]
Support for this new argument has been merged and will release with version 3.3.0 of the Terraform AWS Provider, next week. Thanks to @shuheiktgw for the implementation. 👍
